backend/pkg/api/handlers/auth.go

```diff
@@ -715,45 +719,42 @@ func GetUserIdStoreMiddleware(userIdFunc func(r *http.Request) (uint64, error))
 }
 
 // returns a middleware that checks if user has access to dashboard when a primary id is used
-// expects a userIdFunc to return user id, probably GetUserIdBySession or GetUserIdByApiKey
-func (h *HandlerService) GetVDBAuthMiddleware(userIdFunc func(r *http.Request) (uint64, error)) func(http.Handler) http.Handler {
-	return func(next http.Handler) http.Handler {
-		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-			var err error
-			dashboardId, err := strconv.ParseUint(mux.Vars(r)["dashboard_id"], 10, 64)
-			if err != nil {
-				// if primary id is not used, no need to check access
-				next.ServeHTTP(w, r)
-				return
-			}
-			// primary id is used -> user needs to have access to dashboard
+func (h *HandlerService) VDBAuthMiddleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		var err error
+		dashboardId, err := strconv.ParseUint(mux.Vars(r)["dashboard_id"], 10, 64)
+		if err != nil {
+			// if primary id is not used, no need to check access
+			next.ServeHTTP(w, r)
+			return
+		}
+		// primary id is used -> user needs to have access to dashboard
 
-			userId, err := userIdFunc(r)
-			if err != nil {
-				handleErr(w, err)
-				return
-			}
-			// store user id in context
-			ctx := r.Context()
-			ctx = context.WithValue(ctx, ctxUserIdKey, userId)
-			r = r.WithContext(ctx)
+		userId, err := GetUserIdByContext(r)
+		if err != nil {
+			handleErr(w, err)
+			return
+		}
```

Comment:
Would use the same check in `ManageViaApiCheckMiddleware` (instead of `r.Context().Value(ctxUserIdKey).(uint64)`)